addon/components/admin-fields/admin-field-boolean.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember'
+import layout
+  from 'ember-admin/templates/components/admin-fields/admin-field-boolean'
+
+const {
+  A,
+  Component,
+  computed,
+  isEmpty,
+  computed: { notEmpty }
+} = Ember
+
+export default Component.extend({
+  layout,
+  tagName: 'select',
+  attributeBindings: ['dataTest:data-test'],
+  dataTest: 'admin-field-boolean',
+
+  data: A([true, false]),
+  value: null,
+  boolValue: computed('value', function() {
+    const value = this.get('value')
+
+    if (isEmpty(value)) {
+      return null
+    }
+
+    return value || value === 'true'
```

Comment:
Good point. If this was the string `'false'` this would also misbehave since it would return the string `false` from this function instead of the boolean.